### DISCUSSION TITLE
Use Data Source Story Questions

### DISCUSSION THREAD

**Discussion by engrpatel21** [glideapps/glide-data-grid#412] [Q&A] [answered]
Hi I have some questions on the use data source example in the story book. I read through the code and understand what you are doing. Is the difference here that you are using faker js to generate the data for you, and you are setting the data the with:

```ts
   if (cache.current[key] === undefined) {
                cache.current[key] = faker.name.firstName() + " " + faker.name.lastName();
            }
            const d = cache.current[key];
```
Is this supposed to mimic getting data from a database? Can async/await be used here to fetch data?

**Comment by jassmith** [comment, marked as answer]
Check out the use-async-data-source instead if you need a remote data fetcher.
